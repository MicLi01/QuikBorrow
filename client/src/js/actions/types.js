export const GET_ERRORS = "GET_ERRORS";
export const USER_LOADING = "USER_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SEARCH = "SEARCH";
export const GET_USER_PROFILE = "GET_USER_PROFILE";

// Products
export const FETCH_PRODUCTS = "fetch_products";
export const FETCH_PRODUCT = "fetch_product";
export const BORROW_PRODUCT_FETCH = 'BORROW_PRODUCT_FETCH';
export const BORROW_PRODUCT_FINISHED = 'BORROW_PRODUCT_FINISHED';

