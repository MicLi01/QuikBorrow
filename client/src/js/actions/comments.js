import axios from "axios";
import { CREATE_COMMENT, GET_COMMENTS } from './types';

/**
 * @typedef {import('redux').Dispatch} DispatchFunction
 */

/**
 * Create a comment
 * 
 * @param {Object} comment - The Comment object to create
 * @returns {DispatchFunction}
 */
export const createComment = (comment) => dispatch => {
  return axios
    .post("/api/comments", comment)
    .then(res => {
    	console.log(res);
    	console.log(comment);
    })
    .catch(err => {
    	console.log(err);
  		dispatch({
  			type: CREATE_COMMENT.ERROR,
  			payload: err.response.data
  		})
    });
};

/**
 *  Get comments associated with product
 *  @param {String} id - object id of product
 *  @returns {DispatchFunction} dispatch - used to dispatch actions
 */
export const getComments = (id) => dispatch => {
  return axios
    .get(`/api/comments/${id}`)
    .then(res => {
      dispatch({
        type: GET_COMMENTS.FINISHED,
        payload: res.data
      });
    })
    .catch(err => {
      dispatch({
        type: GET_COMMENTS.ERROR,
        payload: err.response.data
      })
    });
};