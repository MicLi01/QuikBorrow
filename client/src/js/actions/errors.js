import { CLEAR_ALL_ERRORS } from './types'

/**
 * Clears all error data in error state for all error types
 */
 export const clearAllErrors = () => {
     return { type: CLEAR_ALL_ERRORS }
 }
