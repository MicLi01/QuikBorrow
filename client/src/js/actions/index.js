import axios from "axios";
import setAuthToken from "../utils/setAuthToken";
import jwt_decode from "jwt-decode";
import { FETCH_PRODUCTS, FETCH_PRODUCT, SET_CURRENT_USER, USER_LOADING, SEARCH_PRODUCTS, GET_USER, REGISTER_USER, LOGIN_USER } from "./types";
export * from './products';

// Fetch all products
export const fetchProducts = () => {
  return async function(dispatch) {
    const res = await axios.get("/api/items");
    dispatch({ type: FETCH_PRODUCTS, payload: res.data });
  };
};

// Get one product
export const fetchProduct = itemId => {
  return async function(dispatch) {
    const res = await axios.get(`/api/items/${itemId}`);
    dispatch({ type: FETCH_PRODUCT, payload: res.data });
  };
};

/**
 *  Search by query for items in database by calling {@link Action} to backend
 *  @param {String} query - query string to search
 *  @returns {Function} dispatch - used to dispatch actions
 */
export const searchProducts = query => 
  /**
   *  @param {Function} dispatch - used to dispatch actions
   *  @returns {Promise<Object>} - response from an {@link Action}
   */
  dispatch => {
    return axios
        .get("/api/items/search", { params: { param: query} })
        .then(res => {
          dispatch({ type: SEARCH_PRODUCTS.FETCH, payload: res.data });
        })
        .catch(err => {
          dispatch({
            type: SEARCH_PRODUCTS.ERROR,
            payload: err.response.data
          })
        });
  };

// Register User
export const registerUser = (userData, history) => dispatch => {
  return axios
    .post("/api/users/register", userData)
    .then(res => history.push("/login")) // re-direct to login on successful register
    .catch(err =>
      dispatch({
        type: REGISTER_USER.ERROR,
        payload: err.response.data
      })
    );
};

// Login - get user token
export const loginUser = userData => dispatch => {
  return axios
    .post("/api/users/login", userData)
    .then(res => {
      // Save to localStorage
      // Set token to localStorage
      const { token } = res.data;
      localStorage.setItem("jwtToken", token);
      // Set token to Auth header
      setAuthToken(token);
      // Decode token to get user data
      const decoded = jwt_decode(token);
      // Set current user
      dispatch(setCurrentUser(decoded));
    })
    .catch(err =>
      dispatch({
        type: LOGIN_USER.ERROR,
        payload: err.response.data
      })
    );
};

// Set logged in user
export const setCurrentUser = decoded => {
  return {
    type: SET_CURRENT_USER,
    payload: decoded
  };
};

// User loading
export const setUserLoading = () => {
  return {
    type: USER_LOADING
  };
};

// Log user out
export const logoutUser = () => dispatch => {
  // Remove token from local storage
  localStorage.removeItem("jwtToken");
  // Remove auth header for future requests
  setAuthToken(false);
  // Set current user to empty object {} which will set isAuthenticated to false
  dispatch(setCurrentUser({}));
};

export const getUserProfile = userId => {
  return async function(dispatch) {
    try {
    const res = await axios.get(`/api/users/${userId}`);
    dispatch({ type: GET_USER.FINISHED, payload: res.data });
  }
  catch(error) {
    console.log(error)
    dispatch({
      type: GET_USER.ERROR, payload: error
      });
    }
  }
};
