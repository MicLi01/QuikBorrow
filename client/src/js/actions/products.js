import axios from "axios";
import {
    FETCH_PRODUCTS,
    FETCH_PRODUCT,
    SEARCH_PRODUCTS,
    REQUEST_BORROW_PRODUCT,
    FETCH_TRANSACTIONS
} from './types'

/**
 * @typedef {import('redux').Dispatch} DispatchFunction
 */

/**
 * Borrow an item via fetch
 * 
 * @param {Object} item - The Item object to borrow
 * @param {string} msg - The message for the lender
 * @returns {DispatchFunction}
 */
export const requestBorrowProductFetch = (item, msg) => async (dispatch, getState) => {
    const state = getState()
    const body = {
        borrowerId: state.auth.user.id,
        lenderId: item.user,
        itemId: item._id,
        msg
    }
    return axios.post(`/api/transactions`, body)
        .then(({ data }) => {
            dispatch({ type: REQUEST_BORROW_PRODUCT.FINISHED, payload: data })
        })
        .catch(err => {
            dispatch({ type: REQUEST_BORROW_PRODUCT.ERROR, payload: err.response.data })
        })
}

/**
 * Fetch all products
 * @returns {DispatchFunction}
 */
export const fetchProducts = () => {
  return async function(dispatch) {
    const res = await axios.get("/api/items");
    dispatch({ type: FETCH_PRODUCTS, payload: res.data });
  };
};

/**
 * Get one product
 * 
 * @param {Object} itemId - The id of the item to retrieve
 * @returns {DispatchFunction}
 */
export const fetchProduct = itemId => {
  return async function(dispatch) {
    const res = await axios.get(`/api/items/${itemId}`);
    dispatch({ type: FETCH_PRODUCT, payload: res.data });
  };
};

/**
 *  Search by query for items in database by calling {@link Action} to backend
 *  @param {String} query - query string to search
 *  @returns {Function} dispatch - used to dispatch actions
 */
export const searchProducts = query => 
  /**
   *  @param {Function} dispatch - used to dispatch actions
   *  @returns {Promise<Object>} - response from an {@link Action}
   */
  dispatch => {
    return axios
        .get("/api/items/search", { params: { param: query} })
        .then(res => {
          dispatch({ type: SEARCH_PRODUCTS.FINISHED, payload: res.data });
        })
        .catch(err => {
          dispatch({
            type: SEARCH_PRODUCTS.ERROR,
            payload: err.response.data
          })
        });
  };

  /**
 * Get all current user transactions
 * 
 * @param {Object} id - The user id
 * @returns {Function} dispatch - used to dispatch actions
 */

 export const fetchTransactions = (id) => 
  /**
   *  @param {Function} dispatch - used to dispatch actions
   *  @returns {Promise<Object>} - response from an {@link Action}
   */
  dispatch => {
    return axios.post('/api/transactions/getTransactions', {userId : id})
        .then(res => dispatch({ type: FETCH_TRANSACTIONS.FINISHED, payload: res.data }))
        .catch(err => dispatch({ type: FETCH_TRANSACTIONS.ERROR, payload: err.response.data }))
 }

