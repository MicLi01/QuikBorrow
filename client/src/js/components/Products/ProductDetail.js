import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions";

class ProductDetail extends Component {
    componentDidMount() {
        this.props.fetchProduct(this.props.match.params.productId);
    }

    // Just some initial rendering to make sure it works before styling
    renderDetail() {
        return (
            <div>
                <h3>{this.props.product.name}</h3>
                <h3>{this.props.product.description}</h3>
                <h3>{this.props.product._id}</h3>
                <h3>{this.props.product.availability ? 'true' : 'false'}</h3>
                <h3>{this.props.product.user}</h3>
            </div>
        )
    }
    
    render() {
        return <div>{this.renderDetail()}</div>
    }
}

function mapStateToProps(state) {
    console.log(state);
    return { product : state.product }
}

export default connect(mapStateToProps, actions)(ProductDetail);