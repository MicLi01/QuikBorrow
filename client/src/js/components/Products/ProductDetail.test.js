import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ProductDetail } from './ProductDetail';
import { fetchProduct } from "../../actions/index";

Enzyme.configure({ adapter: new Adapter() });
jest.mock('../../actions/index');

jest.mock("react-redux", () => {
  return {
    connect: (mapStateToProps, mapDispatchToProps) => (
      ReactComponent
    ) => ReactComponent
  };
});

jest.mock('../Comments/CommentList', () => ()=> <div id="mockContainer">
   mockContainer
</div>);

describe('Product Detail', () => {
	const param = {
		product: {
			name: 'test',
			description: 'desc',
			_id: '0',
			availability: false,
			user: 'user'
		},
		match: {
			params: {
				productId: 0
			}
		}
	}

	test('renders', () => {
		const auth = {
			user: {}
		}
		const component = renderer.create(
			<ProductDetail product = {param.product} match = {param.match} fetchProduct = {fetchProduct} auth={auth} />
		).toJSON();
		
		expect(component).toMatchSnapshot();
	});
});