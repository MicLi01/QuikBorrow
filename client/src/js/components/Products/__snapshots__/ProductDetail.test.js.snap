// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Detail renders 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root sc-bdVaJa jFJDry MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root sc-bwzfXH fQtfNt"
      style={
        Object {
          "backgroundImage": "url(\\"https://patch.com/img/cdn/users/1142384/2013/09/raw/77d3e8242e7562885116ebff68689271.jpg\\")",
        }
      }
    />
    <div
      className="sc-htpNat fNviHe"
    >
      <div
        className="sc-ifAKCX fOGIFb"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          test
        </h4>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            style={
              Object {
                "color": "grey",
              }
            }
          >
            By: 
            user
          </span>
        </p>
      </div>
      <div
        className="sc-ifAKCX fOGIFb"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
        >
          <span
            style={
              Object {
                "color": "#FFD700",
              }
            }
          >
            Price: 
            $100
          </span>
        </h6>
        <span
          aria-label={null}
          className="MuiRating-root MuiRating-sizeSmall"
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role={null}
        >
          <span
            className="MuiRating-decimal"
          >
            <label
              className="MuiRating-label"
              htmlFor="half-rating-0-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
                style={
                  Object {
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": "0%",
                    "zIndex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                0.5 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-0-5"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={0.5}
            />
            <label
              className="MuiRating-label"
              htmlFor="half-rating-1"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
                style={Object {}}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                1 Star
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-1"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={1}
            />
          </span>
          <span
            className="MuiRating-decimal"
          >
            <label
              className="MuiRating-label"
              htmlFor="half-rating-1-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
                style={
                  Object {
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": "0%",
                    "zIndex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                1.5 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-1-5"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={1.5}
            />
            <label
              className="MuiRating-label"
              htmlFor="half-rating-2"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
                style={Object {}}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                2 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-2"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={2}
            />
          </span>
          <span
            className="MuiRating-decimal"
          >
            <label
              className="MuiRating-label"
              htmlFor="half-rating-2-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
                style={
                  Object {
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": "50%",
                    "zIndex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                2.5 Stars
              </span>
            </label>
            <input
              checked={true}
              className="MuiRating-visuallyhidden"
              id="half-rating-2-5"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={2.5}
            />
            <label
              className="MuiRating-label"
              htmlFor="half-rating-3"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
                style={Object {}}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                3 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-3"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={3}
            />
          </span>
          <span
            className="MuiRating-decimal"
          >
            <label
              className="MuiRating-label"
              htmlFor="half-rating-3-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
                style={
                  Object {
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": "0%",
                    "zIndex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                3.5 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-3-5"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={3.5}
            />
            <label
              className="MuiRating-label"
              htmlFor="half-rating-4"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
                style={Object {}}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                4 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-4"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={4}
            />
          </span>
          <span
            className="MuiRating-decimal"
          >
            <label
              className="MuiRating-label"
              htmlFor="half-rating-4-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
                style={
                  Object {
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": "0%",
                    "zIndex": 1,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                4.5 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-4-5"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={4.5}
            />
            <label
              className="MuiRating-label"
              htmlFor="half-rating-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
                style={Object {}}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                5 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="half-rating-5"
              name="half-rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={5}
            />
          </span>
        </span>
      </div>
      <div
        style={
          Object {
            "marginTop": "30px",
          }
        }
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          <span
            style={
              Object {
                "color": "grey",
              }
            }
          >
            Description
          </span>
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          desc
        </p>
      </div>
      <div
        style={
          Object {
            "marginTop": "30px",
          }
        }
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          <span
            style={
              Object {
                "color": "grey",
              }
            }
          >
            Availability
          </span>
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span>
            Unavailable (currently being lended out)
          </span>
        </p>
      </div>
      <div
        style={
          Object {
            "marginTop": "30px",
          }
        }
      >
        <div
          className="sc-ifAKCX fOGIFb"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Add to Cart
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
              Add to Wishlist
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="sc-bxivhb gNLXtS"
    >
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiBox-root MuiBox-root-126"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="body"
              id="body-label"
            >
              Type a comment
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="body"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-187 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-188"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <span
            aria-label={null}
            className="MuiRating-root"
            id="rating"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role={null}
          >
            <label
              className="MuiRating-label"
              htmlFor="rating-1"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                1 Star
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="rating-1"
              name="rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={1}
            />
            <label
              className="MuiRating-label"
              htmlFor="rating-2"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                2 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="rating-2"
              name="rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={2}
            />
            <label
              className="MuiRating-label"
              htmlFor="rating-3"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                3 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="rating-3"
              name="rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={3}
            />
            <label
              className="MuiRating-label"
              htmlFor="rating-4"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                4 Stars
              </span>
            </label>
            <input
              checked={false}
              className="MuiRating-visuallyhidden"
              id="rating-4"
              name="rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={4}
            />
            <label
              className="MuiRating-label"
              htmlFor="rating-5"
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-visuallyhidden"
              >
                5 Stars
              </span>
            </label>
            <input
              checked={true}
              className="MuiRating-visuallyhidden"
              id="rating-5"
              name="rating"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value={5}
            />
          </span>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          id="comment"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Comment
          </span>
        </button>
      </form>
      <div
        id="mockContainer"
      >
        mockContainer
      </div>
    </div>
  </div>
</div>
`;
